### ISSUE TITLE
Directory groups can apply to indirect as well as direct members

### ISSUE THREAD

**Issue by bhunut-adobe** [adobe-apiplatform/user-sync.py#182]
When following this [guide](https://github.com/adobe-apiplatform/user-sync.py/blob/613cd248720e7501d78c3834367cc9360c833681/docs/user-manual/index.md#working-with-nested-directory-groups-in-active-directory),
UST returned the following error
Multiple LDAP groups found for: AdobeCCETest-All-App

Current Group Structure
```
AdobeCCETest-All-App
	AdobeCCETest-South-Pole
	      jadobe2@adobeccetest.com
```

Log file:
[2017-05-10.txt](https://github.com/adobe-apiplatform/user-sync.py/files/989982/2017-05-10.txt)

**Comment by phil-levy** [comment]
Chester, can your team take a look at this?  It may be that the code doesn't quite handle this nested case properly.

**Comment by chesterl86** [comment]
Sure can.

**Comment by ianmak** [comment]
user-sync doesn't seem to be processing the results of the nested group filter properly at the moment. We are currently planning to refactor this feature to process groups properly, and may support nested groups using a different approach rather than through the AD nested group filter.

**Comment by bhunut-adobe** [comment]
@adobeDan - I just read your comment on the PR. Will this still be included in v2.1 release?

**Comment by adobeDan** [comment]
No, this will not be in 2.1.1.  We thought this was an included feature in 2.0, but it turns out it wasn't.  That makes it an enhancement request, not a bug.  It will go into a 2.2.  Just for you, @bhunut-adobe :), I will try to get 2.2rc1 out next week so you can validate the approach and give it to whatever customer is deperate for it.  But no promises about when 2.2 itself will come.
